lib/shoulda/matchers/active_model/comparison_matcher.rb

```diff
@@ -0,0 +1,158 @@
+require 'active_support/core_ext/module/delegation'
+
+module Shoulda
+  module Matchers
+    module ActiveModel
+      # @private
+      class ComparisonMatcher < ValidationMatcher
+        ERROR_MESSAGES = {
+          :> => {
+            label: :greater_than,
+            assertions: [false, false, true],
+          },
+          :>= => {
+            label: :greater_than_or_equal_to,
+            assertions: [false, true, true],
+          },
+          :< => {
+            label: :less_than,
+            assertions: [true, false, false],
+          },
+          :<= => {
+            label: :less_than_or_equal_to,
+            assertions: [true, true, false],
+          },
+          :== => {
+            label: :equal_to,
+            assertions: [false, true, false],
+          },
+          :!= => {
+            label: :other_than,
+            assertions: [true, false, true],
+          },
+        }.freeze
+
+        delegate :failure_message, :failure_message_when_negated, to: :submatchers
+
+        def initialize(matcher, value, operator)
+          super(nil)
+          unless matcher.respond_to? :diff_to_compare
+            raise ArgumentError, 'matcher is invalid'
+          end
+
+          @matcher = matcher
+          @value = value
+          @operator = operator
+          @message = ERROR_MESSAGES[operator][:label]
+        end
+
+        def simple_description
+          description = ''
+
+          if expects_strict?
+            description << ' strictly'
+          end
+
+          description +
+            "disallow :#{attribute} from being a number that is not " +
+            "#{comparison_expectation} #{@value}"
+        end
+
+        def for(attribute)
+          @attribute = attribute
+          self
+        end
+
+        def with_message(message)
+          @expects_custom_validation_message = true
+          @message = message
+          self
+        end
+
+        def expects_custom_validation_message?
+          @expects_custom_validation_message
+        end
+
+        def matches?(subject)
+          @subject = subject
+          build_option_value
+          submatchers.matches?(subject)
+        end
+
+        def does_not_match?(subject)
+          @subject = subject
+          build_option_value
+          submatchers.does_not_match?(subject)
+        end
+
+        def comparison_description
+          "#{comparison_expectation} #{@value}"
+        end
+
+        def submatchers
+          @_submatchers ||= NumericalityMatchers::Submatchers.new(build_submatchers)
+        end
+
+        private
+
+        def build_submatchers
```

Comment:
Hmm, it looks like we have `@matcher` and we have `submatchers`. Should we rename these so that people won't get confused by them?

Reply:
Would you happen to have any idea here? Maybe calling `submatchers` as `comparison_submatchers`?